apps/src/templates/progress/BubbleFactory.jsx

```diff
@@ -82,15 +91,43 @@ DiamondContainer.propTypes = {
   children: PropTypes.node
 };
 
-export function LinkWrapper({url, children}) {
+export function BubbleLink({url, onClick, children}) {
   return (
-    <a href={url} style={bubbleStyles.link}>
+    <a href={url} onClick={onClick} className="progress-bubble-link">
       {children}
     </a>
   );
 }
-LinkWrapper.propTypes = {
-  url: PropTypes.string.isRequired,
+BubbleLink.propTypes = {
+  url: PropTypes.string,
+  onClick: PropTypes.func,
+  children: PropTypes.element.isRequired
+};
+
+export function BubbleTooltip({level, children}) {
+  let tooltipText = level.isSublevel
```

Comment:
I find these ternaries hard to read. Is there a way we can write this so it's easier to understand at a glance what's going on?

Reply:
👍